extensions/sql-migration/src/wizard/databaseBackupPage.ts

```diff
@@ -525,89 +525,105 @@ export class DatabaseBackupPage extends MigrationWizardPage {
+				this.migrationStateModel.databaseBackup.storageAccount = this.migrationStateModel.getStorageAccount(0);
+			} catch (error) {
+				console.log(error);
+			} finally {
+				this._networkShareContainerStorageAccountDropdown.loading = false;
+			}
 		}
 	}
 
 	private async loadFileShareStorageDropdown(): Promise<void> {
-		this._fileShareStorageAccountDropdown.loading = true;
-		this._fileShareFileShareDropdown.loading = true;
-		try {
-			this._fileShareStorageAccountDropdown.values = await this.migrationStateModel.getStorageAccountValues(this.migrationStateModel.databaseBackup.subscription);
-			this.migrationStateModel.databaseBackup.storageAccount = this.migrationStateModel.getStorageAccount(0);
-		} catch (error) {
-			this.migrationStateModel._fileShares = undefined!;
-		} finally {
-			await this.loadFileShareDropdown();
-			this._fileShareStorageAccountDropdown.loading = false;
-			this._fileShareFileShareDropdown.loading = false;
-
+		if (!this.migrationStateModel.databaseBackup.storageAccount) {
+			this._fileShareStorageAccountDropdown.loading = true;
+			this._fileShareFileShareDropdown.loading = true;
+			try {
+				this._fileShareStorageAccountDropdown.values = await this.migrationStateModel.getStorageAccountValues(this.migrationStateModel.databaseBackup.subscription);
+				this.migrationStateModel.databaseBackup.storageAccount = this.migrationStateModel.getStorageAccount(0);
+			} catch (error) {
+				console.log(error);
+			} finally {
+				this.migrationStateModel._fileShares = undefined!;
+				await this.loadFileShareDropdown();
+				this._fileShareStorageAccountDropdown.loading = false;
+				this._fileShareFileShareDropdown.loading = false;
+			}
 		}
 	}
 
 	private async loadblobStorageDropdown(): Promise<void> {
-		this._blobContainerStorageAccountDropdown.loading = true;
-		this._blobContainerBlobDropdown.loading = true;
-		try {
-			this._blobContainerStorageAccountDropdown.values = await this.migrationStateModel.getStorageAccountValues(this.migrationStateModel.databaseBackup.subscription);
-			this.migrationStateModel.databaseBackup.storageAccount = this.migrationStateModel.getStorageAccount(0);
-		} catch (error) {
-			this.migrationStateModel._blobContainers = undefined!;
-		} finally {
-			await this.loadBlobContainerDropdown();
-			this._blobContainerStorageAccountDropdown.loading = false;
+		if (!this.migrationStateModel.databaseBackup.storageAccount) {
+			this._blobContainerStorageAccountDropdown.loading = true;
 			this._blobContainerBlobDropdown.loading = true;
+			try {
+				this._blobContainerStorageAccountDropdown.values = await this.migrationStateModel.getStorageAccountValues(this.migrationStateModel.databaseBackup.subscription);
+				this.migrationStateModel.databaseBackup.storageAccount = this.migrationStateModel.getStorageAccount(0);
+			} catch (error) {
+				console.log(error);
+			} finally {
+				this.migrationStateModel._blobContainers = undefined!;
+				await this.loadBlobContainerDropdown();
```

Comment:
What about if an error occurs while loading the blob containers dropdown? This should probably be in its own try/catch/finally

Same goes for other similar logic in the other functions

Reply:
Each of these function have their own try catch. So, an error should be thrown by this function.

Reply:
I'm saying that currently if an error occurs you won't be resetting the loading states back - are you implying that that's handled by the callers instead?

(although looking through the logic I also have some questions - for example you set blobContainerBlobDropdown.loading to true on both line 586 and 596. Is that intentional?)

Reply:
The errors are handled by the called function. So, the called function is never going to throw an error. 

For the loading questions, that is pure UX:
For example, when the storage account is loading, the blob container should load as well.